### PR TITLE
Fix SSO for new users

### PR DESCRIPTION
**Pull request by timolegros**
## Link to Issue
Closes: #8495 

## Description of Changes
- Removes some unnecessary code 

## "How We Fixed It"
- Rather than redirecting to the sign-in modal again after selecting `create account` and signing in, sign the user in.
- If a new user uses `sign-in` instead of `create-account` just create an account for them and sign-in rather than completing the magic flow and ending in a non-signed in state.

## Test Plan
- Create/get a new social account that you have _never_ used to sign in (this must be an account that you have not used locally or on any Heroku app since Magic saves emails regardless of the environment used.

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

File: packages/commonwealth/client/scripts/controllers/app/login.ts [hunk 1]
```diff
@@ -515,25 +515,26 @@ export async function handleSocialLoginCallback({
...
+  // const isAttemptingToConnectAddressToCommunity =
+  //   app.isLoggedIn() && app.activeChainId();
+
+  // if (
+  //   isAddressNew &&
+  //   !isAttemptingToConnectAddressToCommunity &&
+  //   returnEarlyIfNewAddress
+  // ) {
+  //   return { address: magicAddress, isAddressNew };
+  // }
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/controllers/app/login.ts:537]: removing this would break new account signup/onboard flow for users

**Review by masvelio — DISMISSED** [review]
we need more in depth investigation and if after discussions with product/design there are still issues, we need solution that fix this, but do not break another part of the flow

**Comment by timolegros** [comment]
> we need more in depth investigation and if after discussions with product/design there are still issues, we need solution that fix this, but do not break another part of the flow

The goal is obviously not to break another part of the flow and I only put this up for testing/visibility hence why it is in draft. New user sign in _is broken_.

**Review by burtonator — APPROVED** [review]

**Review by timolegros — COMMENTED** [review]
This fixed the 'new user clicks on sign in flow' but I think it broke the create account flow. If I try to create an account as a new user I get to this screen:
image


Shouldn't this be skipped and go straight to 
image

**Comment by mzparacha** [comment]
@timolegros - I am not able to reproduce with metamask and google sso


https://github.com/user-attachments/assets/7aa71216-69b2-442e-a528-7f7787a690de

**Comment by timolegros** [comment]
Ok I think I fat fingered it --- I approve the changes. Unfortunately, I can't approve them officially on GitHub because I originally open this PR lol

**Review by mzparacha — APPROVED** [review]

**Comment by mzparacha** [comment]
@ilijabojanovic can you please share the error message (you can click on the link in the console, and view the "response" tab in network tab)

image

**Comment by timolegros** [comment]
@ilijabojanovic if you used the same email on Discord and Google social accounts than a `500 validation error` is to be expected. This is a separate issue related to email uniqueness in the database.

**Pull request merged**